boudicca.base/enricher/src/main/kotlin/base/boudicca/enricher/service/osm/OsmEnricher.kt

```diff
@@ -0,0 +1,193 @@
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_ADDRESS_PROPERTY,
+            buildAddress(tags)
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_CITY_PROPERTY,
+            tags["addr:city"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_DESCRIPTION_PROPERTY,
+            tags["amenity"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_CONTACT_EMAIL_PROPERTY,
+            tags["contact:email"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_CONTACT_PHONE_PROPERTY,
+            tags["contact:phone"]?.asText()
+        )
+        try {
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_URL_PROPERTY,
+                UrlUtils.parse(tags["contacts:website"]?.asText())
+            )
+        } catch (_: IllegalArgumentException) {
+            logger.info("Could not map ${tags["contacts:website"].asText()} to URI")
+        }
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_WIKIPEDIA_PROPERTY,
+            tags["wikipedia"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_WIKIDATA_PROPERTY,
+            tags["wikidata"]?.asText()
+        )
+
+        return builder.build()
+    }
+
+    private fun getOrSearchOsmId(event: StructuredEvent): String? {
+        // TODO maybe think about taking all osmIds and not only the first :)
+        val osmIds = event.getProperty(SemanticKeys.LOCATION_OSM_ID_PROPERTY)
+        val osmId = if (osmIds.isNotEmpty()) {
+            osmIds.first().second
+        } else {
+            searchOsmId(event)
+        }
+        return osmId
+    }
+
+    private fun searchOsmId(event: StructuredEvent): String? {
+        val eventName = event
+            .getProperty(SemanticKeys.LOCATION_NAME_PROPERTY)
+            .map(Pair<Key, String>::second)
+            .joinToString(separator = ",") { it.encodeURL() }
+
+        val query = """https://nominatim.openstreetmap.org/search?q=$eventName&format=json&limit=1"""
+        val request = HttpRequest.newBuilder()
+            .uri(URI.create(query))
+            .GET()
+            .header("User-Agent", Constants.USER_AGENT)
+            .build()
+
+        val response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString()).body()
+        val json = jacksonObjectMapper().readTree(response) as ArrayNode
+
+        return if (json.isEmpty) null
+        else json.first()["osm_id"].asText()
+    }
+
+    private fun buildAddress(tags: JsonNode): String {
+        val street = (tags["addr:street"]?.asText() ?: "") + " " + (tags["addr:housenumber"]?.asText() ?: "")
```

Comment:
we should trim this string, otherwise there could be a leading or trailing whitespace